hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityConverter.java

```diff
@@ -58,17 +63,22 @@
   }
 
   Single<EntityResultSet> convert(
-      EntityRequest request, EntitiesResponse response, BaselineEntitiesResponse baselineResponse) {
+      EntityRequest request,
+      EntitiesResponse response,
+      BaselineEntitiesResponse baselineResponse,
+      Map<String, String> labelsMap) {
```

Comment:
I am guessing the value in the map is the label key. Instead, the value in the map should be the entire server Label response

Reply:
it is <label-id> to <label-key> map. Since the response is not per entity, I had converted it once and passed it as an argument.

Reply:
anyways the logic is now moved to converter